SYCL/Matrix/element_wise_int8_sumLocalRow.cpp

```diff
@@ -0,0 +1,192 @@
+//==------- element_wise_int8_sumLocalRow.cpp  - DPC++ joint_matrix---------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: matrix
+
+// RUN: %clangxx -fsycl %s -o %t.out -DSYCL_EXT_ONEAPI_MATRIX_VERSION=4
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+// Caclulate sum of rows for a matrix type A.
+#include <iostream>
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental::matrix;
+
+#define SG_SZ 16
+
+#define TM 8
+#define TN SG_SZ
+#define TK 32
+
+template <typename T, size_t NUM_ROWS, size_t NUM_COLS> struct big_matrix {
+public:
+  T *mat;
+
+public:
+  T *get_data() { return mat; }
+  void set_data(T *data) { mat = data; }
+  big_matrix(T *data) : mat(data) {}
+};
+
+template <typename T, size_t M, size_t K>
+void sum_rows_ref(host_accessor<T, 2, access::mode::read_write> A,
+                  host_accessor<int, 1, access::mode::read_write> sum_rows) {
+  int sum_rows_ref[M] = {0};
+  for (size_t i = 0; i < M; i++) {
+    for (size_t j = 0; j < K; j++) {
+      sum_rows_ref[i] += A[i][j];
+    }
+    auto diff = sum_rows[i] - sum_rows_ref[i];
+    assert(std::fabs(static_cast<int>(diff)) <=
+           std::numeric_limits<int>::epsilon());
+  }
+}
+
+template <typename T, size_t M, size_t K>
+void matrix_sum_rows(queue q, big_matrix<T, M, K> &A, nd_range<2> &r) {
+  buffer<int8_t, 2> bufA(A.get_data(), range<2>(M, K));
+  // size of vector is known because SG size of set by the user in this case
+  int sum_rows[M] = {0};
+  buffer<int> sum_rows_v(sum_rows, M); // there are total of M rows
+  q.submit([&](handler &cgh) {
+     accessor accA{bufA, cgh};
+     auto v = sum_rows_v.template get_access<access::mode::read_write>(cgh);
```

Comment:
Why using to different styles for the same thing?

Reply:
If I use   `accessor v{sum_rows_v, cgh};`, will the access::mode be set to `read_write` by default?  This style of getting access was inspired by one of the test-suite test cases, which was using `atomic_ref`

Reply:
Yes, it will be.